sdk/mgmtcommon/ClientRuntime/ClientRuntime/Serialization/JsonConverterHelper.cs

```diff
@@ -113,19 +115,24 @@ public static string GetPropertyName(this JsonProperty property, out string[] pa
 
             if (!string.IsNullOrEmpty(propertyName))
             {
-                string[] hierarchy = Regex.Split(propertyName, @"(?<!\\)\.")
-                    .Select(p => p?.Replace("\\.", ".")).ToArray();
+                string[] hierarchy = splitCompiledRegex.Split(propertyName);
```

Comment:
As this func is frequently invoked, so consider not use Regex but just a single loop to complete both split and replace.

Reply:
what I understand from the code the split with regex is used to split by '.' and skip splitting by '\.' as far it's escaped '.'
so not sure how can you in one loop split by '.' with skipping '\.' not using the regex and then replacing '\.' with '.'
if you first split by '.' and then you need to go thru loop and join subsequent array items ending with '\' ..
I guess this solution us much better than it was so I would let it for someone from your team to polish it if needed